### PR TITLE
Migrate from old-style .java provider to JavaInfo.

### PR DESCRIPTION
**Pull request by iirina**
Migrating in preparation of flipping the flag [--incompatible_disallow_legacy_java_provider](https://github.com/bazelbuild/bazel/issues/7598) in bazel.

### CODE REVIEW

File: jmh/jmh.bzl [hunk 1]
```diff
@@ -77,7 +77,7 @@ def jmh_repositories(maven_servers = ["http://central.maven.org/maven2"]):
 
 def _scala_construct_runtime_classpath(deps):
     scala_targets = [d.scala for d in deps if hasattr(d, "scala")]
-    java_targets = [d.java for d in deps if hasattr(d, "java")]
+    java_targets = [d[JavaInfo] for d in deps if hasattr(d, "JavaInfo")]
```
Comment (johnynek) [inline jmh/jmh.bzl]: This was written before scala returned a java provider. I think we can remove the scala targets entirely and just use `targets` in fact we can probably combine this with the loop below that constructs the depset.

File: scala/private/common.bzl [hunk 2]
```diff
@@ -44,8 +44,8 @@ def collect_plugin_paths(plugins):
             paths.append(p)
         elif hasattr(p, "scala"):
             paths.extend([j.class_jar for j in p.scala.outputs.jars])
-        elif hasattr(p, "java"):
-            paths.extend([j.class_jar for j in p.java.outputs.jars])
+        elif hasattr(p, "JavaInfo"):
```
Comment (johnynek) [inline scala/private/common.bzl]: Can we change this to `elif JavaInfo in d`? Similar comment about removing the scala branch.

**Comment by johnynek** [comment]
Thanks for the PR!

File: jmh/jmh.bzl (hunk 1 above)
Comment (iirina) [reply jmh/jmh.bzl]: Great, done!

File: scala/private/common.bzl (hunk 2 above)
Comment (iirina) [reply scala/private/common.bzl]: Done.

**Comment by johnynek** [comment]
Great! Thank you for this improvement!

**Comment by iirina** [comment]
Thanks for the quick code reviews! \o/

**Pull request merged**